lib/sphinx-preview.js

```diff
@@ -36,13 +43,54 @@ export default {
     }
   },
 
-  getBindingPath() {
+  getCurrProjectPath() {
+    const editor = atom.workspace.getActivePaneItem();
+    const activeFilePath = editor.buffer.file.path;
+
+    if (!activeFilePath) return null;
+
+    for (const path of atom.project.getPaths()) {
+      if (activeFilePath.startsWith(path)) {
+        return path;
+      }
+    }
+    return null;
+  },
+
+  getContainerIdx(serverPath) {
+    if (!(serverPath in this.servers)) {
+      this.servers[serverPath] = this.idxs;
+      this.idxs += 1;
+    }
+
+    return this.servers[serverPath];
+  },
+
+  getContainerName(serverPath) {
+    var hash =  new Hashes.SHA256
```

Comment:
Correct me if I'm wrong, but don't we have access to NodeJS native libraries here? If so, I would prefer usage of the native library https://nodejs.org/api/crypto.html